shenyu-kubernetes-controller/src/main/java/org/apache/shenyu/k8s/parser/DivideIngressParser.java

```diff
@@ -290,15 +292,15 @@ private List<DivideUpstream> parseUpstream(final V1IngressBackend backend, final
                     if (Objects.isNull(addresses) || addresses.isEmpty()) {
                         continue;
                     }
+                    int i = 0;
                     for (V1EndpointAddress address : addresses) {
                         String upstreamIp = address.getIp();
                         String defaultPort = parsePort(backend.getService());
                         if (Objects.nonNull(defaultPort)) {
                             DivideUpstream upstream = new DivideUpstream();
                             upstream.setUpstreamUrl(upstreamIp + ":" + defaultPort);
                             upstream.setWeight(100);
-                            // TODO support config protocol in annotation
-                            upstream.setProtocol("http://");
+                            upstream.setProtocol(Objects.isNull(protocol) ? "http://" : protocol.split(",")[i++]);
```

Comment:
Why is an array?

Reply:
In the yaml file, we need to set up several different Upstreams at the same time, so we will set up the yaml file in the following form:
```yaml
metadata.
  name: ...
  annotations.
    yourAnnotationKey: ... 
      - value1
      - value2
      - value3
```
Then read in [value1,value2,value3], so it's good to use an array to iterate through each of these values

Reply:
The split is used once every cycle, which is too inefficient, and the split should be placed outside the cycle.

Reply:
done

> The split is used once every cycle, which is too inefficient, and the split should be placed outside the cycle.